vjudge/status.py

```diff
@@ -0,0 +1,67 @@
+
+
+status_keywords = ['accept', 'answer', 'compilation', 'compile', 'error',
+    'exceed', 'limit', 'memory', 'runtime', 'time', 'wrong']
+
+# Fetch status every 3 seconds
+# Should run as daemon
+def status_daemon():
+    while True:
+        try:
+            raw_status = requests.get(status_url, timeout=5).text.encode("utf-8")
+            raw_status = json.loads(raw_status)['data']
+
+            refined_status = {}
+            for status in raw_status:
+                sid = status[0]
+                result = status[3]
+                refined_status[sid] = result
+
+            submissions = Submission.objects.filter(status__in=[Submission.WAIT, Submission.JUDGING])
+
+            for submission in [s for s in submissions if s.problem.judge_source == Problem.OTHER and s.other_judge_sid]:
+                status = refined_status.get(submission.other_judge_sid)
+                if status:
+                    status_lower = status.lower()
+                    # If any of the keyword is in the status, write result to submission
+                    for keyword in status_keywords:
+                        if keyword in status_lower:
+                            submission.status = status
```

Comment:
Nothing harm will happen. The choice we provide in the model is like a goal keeper  that prohibit front-end user from entering something we don't want.
If it's insert from back-end, it will be accepted.